app/models/step.rb

```diff
@@ -49,4 +50,8 @@ def batch_link_report_id
   def most_recent_batch
     LinkReport.where(step_id: self.id).last
   end
+
+  def set_step_position
+    update!(position: step_by_step_page.steps.count)
```

Comment:
Would it be better to use `step_by_step_page.steps.maximum(:position) + 1` to be absolutely sure you are incrementing the highest step by one?

Reply:
Hmm I've tried updating as suggested, but this leads to a _lot_ of test failures. I think it's to do with how Rails factories create things.

For example, an assertion that the first step has position 1 works, but an assertion that the next step has position 2 fails (it's actually 3!). Haven't figured out why it's doing that yet.
I also had to modify the code to get the 'position 1' example to work, as `step_by_step_page.steps.maximum(:position)` returns `nil` if there are no existing steps.

I'm inclined to leave this and rely on the existing logic to keep things in order.